packages/expo-branch/android/src/main/java/expo/modules/branch/BranchIntentNotifier.java

```diff
@@ -0,0 +1,29 @@
+package expo.modules.branch;
+
+import android.app.Activity;
+import android.content.Intent;
+
+import org.unimodules.core.interfaces.ActivityEventListener;
+import org.unimodules.core.interfaces.InternalModule;
+
+import java.util.Collections;
+import java.util.List;
+
+import io.branch.rnbranch.RNBranchModule;
+
+public class BranchIntentNotifier implements InternalModule, ActivityEventListener {
+  @Override
+  public void onActivityResult(Activity activity, int requestCode, int resultCode, Intent data) {
+    // do nothing
+  }
+
+  @Override
+  public void onNewIntent(Intent intent) {
+    RNBranchModule.onNewIntent(intent);
```

Comment:
Should this call the superclass's implementation too?
```suggestion
    super.onNewIntent(intent);
    RNBranchModule.onNewIntent(intent);
```

Reply:
We could if `BranchIntentNotifier` was a subclass of another class (obv.). The way I approached this — instead of subclassing `RNBranchModule` and adding the `ActivityEventListener` conformance there, I added this simple class that's just a binding between unimodules realm which lets us listen to `onNewIntent` notifications and `RNBranchModule.onNewIntent` (which is static).

Reply:
I see, the real Activity's onNewIntent will still be called anyway.